lib/ui/platform_dispatcher.dart

```diff
@@ -1943,8 +1943,7 @@ class Locale {
   const Locale(
     this._languageCode, [
     this._countryCode,
-  ]) : assert(_languageCode != null),
-       assert(_languageCode != ''),
+  ]) : assert(_languageCode != ''),
```

Comment:
I wonder whether `assert(_languageCode != '')` or `assert(_languageCode.isNotEmpty)` is more efficient? My guess is that string equals special cases the zero length check anyway, so probably no real difference and it's a debug-only check anyway.

Reply:
@rmacnak-google Do you happen to know?

Reply:
I'd go with `isNotEmpty` from the readability point of view too.

Reply:
Sounds good, since this PR is so giant, I'll send a follow-up PR with that fix. Thanks, @rmacnak-google.

Reply:
Sounds good, since this PR is so giant, I'll send a follow-up PR with that fix. Thanks, @rmacnak-google.

Reply:
Oh, I guess we cannot make that change:

```
platform_dispatcher.dart:446:16: Error: Not a constant expression.
  ])  : assert(_languageCode.isNotEmpty),
               ^^^^^^^^^^^^^
```